pkg/apis/extensions/validation/validation_test.go

```diff
@@ -1300,3 +1300,101 @@ func newInt(val int) *int {
 	*p = val
 	return p
 }
+
+func TestValidateConfigMap(t *testing.T) {
+	newConfigMap := func(name, namespace string, data map[string]string) extensions.ConfigMap {
+		return extensions.ConfigMap{
+			ObjectMeta: api.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+			Data: data,
+		}
+	}
+
+	var (
+		validConfigMap = newConfigMap("validname", "validns", map[string]string{"key": "value"})
+
+		emptyName     = newConfigMap("", "validns", nil)
+		invalidName   = newConfigMap("NoUppercaseOrSpecialCharsLike=Equals", "validns", nil)
+		emptyNs       = newConfigMap("validname", "", nil)
+		invalidNs     = newConfigMap("validname", "NoUppercaseOrSpecialCharsLike=Equals", nil)
+		invalidKey    = newConfigMap("validname", "validns", map[string]string{"a..b": "value"})
+		leadingDotKey = newConfigMap("validname", "validns", map[string]string{".ab": "value"})
+		dotKey        = newConfigMap("validname", "validns", map[string]string{".": "value"})
+		doubleDotKey  = newConfigMap("validname", "validns", map[string]string{"..": "value"})
```

Comment:
a test of `strings.Repeat("a", 253)` which passes and `strings.Repeat("a", 254)` which fails would be nice